app/assets/javascripts/admin/addon/models/permalink.js

```diff
@@ -6,10 +6,14 @@ import discourseComputed from "discourse-common/utils/decorators";
 
 export default class Permalink extends RestModel {
   static findAll(filter) {
-    return ajax("/admin/permalinks.json", { data: { filter } }).then(function (
-      permalinks
-    ) {
-      return permalinks.map((p) => Permalink.create(p));
+    return ajax("/admin/permalinks.json").then(function (permalinks) {
+      let allLinks = permalinks.map((p) => Permalink.create(p));
+
+      let filteredLinks = filter
+        ? allLinks.filter((p) => p.url.includes(filter))
```

Comment:
Is the type of `filter` stable? Asking because:

```javascript
"foo".includes(undefined) // false
"foo".includes("")        // true
```

which could cause inconsistent behaviour depending on state.

Reply:
I think it works because the initial state for `filter` is `null`. In that case `allLinks` are returned without filtering. If I type something into the filter and then delete it, `filter` value will be `""`, which will include everything.

I just realised that before it was filtered on the backend and was returning:
`permalinks.where!("url ILIKE :url OR external_url ILIKE :url", url: "%#{url}%") if url.present?`

I will add OR external_url